control/server.py

```diff
@@ -197,6 +201,21 @@ def serve(self):
             log_level = log_level.upper()
             log_req = pb2.set_spdk_nvmf_logs_req(log_level=log_level, print_level=log_level)
             self.gateway_rpc.set_spdk_nvmf_logs(log_req)
+        
+        self._register_service_map()
+
+    def _register_service_map(self):
+        # show gateway in "ceph status" output
```

Comment:
Instead of adding Rados code in server.py we can just use the existing connection self.omap_state.conn

Reply:
That was my original plan too but I couldn't figure out where the connection is closed, I couldn't find an explicit `shutdown()` call. I've changed the function to use omap_state.conn now. Tested it and it seems to close connection when I remove a gateway - still unsure how.

Reply:
@VallariAg we close the connection in state.py function cleanup_omap().  You can add the shutdown there if you want.

Reply:
Oh okay thanks for sharing, it seems to work well as it is.

Reply:
I guess as we exit anyway there connection is shutdown for us. But let us add an explicit shutdown anyway.